### PR TITLE
Refactor primitives

### PR DESCRIPTION
**Pull request by ianrohde**
Summary of changes
---

1. change `interface` to `type`
2. add `minBreakpoint` primitives
3. add `transition` primitives
4. change `black` value from `#000` to `#11161a`
5. remove `kabab-case` package, inline function instead
6. run npm build script `preversion`
7. bump to `2.4.0`

First step towards completing https://github.com/iFixit/ifixit/issues/40436, with a few improvements thrown in for good measure.

qa_req 0

Merge checklist
---

- [x] Bump the package version by running `npm version [patch | minor | major]` from the command line (if applicable).

> **Note:** In the context of Core Primitives, significant changes to the library or workflow, or removing primitives would be considered a major update, adding or updating primitives would be considered a minor update, and fixing primitives would be considered a patch. Non-code changes (e.g. documentation) do not require a version bump.

### CODE REVIEW

File: bin/build.js [hunk 1]
```diff
@@ -1,8 +1,15 @@
...
 const fs = require("fs");
 const path = require("path");
-const kebabCase = require("kebab-case");
-const { version } = require("../package.json");
 const primitives = require("../index.json");
+const version = process.env.npm_package_version;
+
+function kebabCase(str) {
+  return str.replace(/[A-Z\u00C0-\u00D6\u00D8-\u00DE]/g, function (match) {
+    return "-" + match.toLowerCase();
```
Comment (danielbeardsley) [inline bin/build.js:10]: This doesn't seem quite right.  It'll replace all upper-case chars with `-{char}`

i.e. a string like `ABCD` would transform to `-A-B-C-D`.  Possible in practice we always start with a lowercase letter and never have multiple capitals in a row and this doesn't functionality produce a bug at the moment.

I might do this: `replace(/([a-z])([A-Z]+)/g, (matches) => matches[1] + "-" + matches[2].toLowerCase())

**Comment by danielbeardsley** [comment]
I don't really understand the distinction of types vs interfaces, but this looks reasonable. CR :+1:

File: bin/build.js (hunk 1 above)
Comment (ianrohde) [reply bin/build.js:10]: I pulled that straight from the [`kebab-case` npm package](https://github.com/joakimbeng/kebab-case/blob/master/index.js). It's been doing exactly what I was looking for when building the package 🤷

**Comment by ianrohde** [comment]
> I don't really understand the distinction of types vs interfaces, but this looks reasonable. CR 👍

It's nested in the parent issue: https://github.com/iFixit/ifixit/pull/40259#discussion_r737827593

**Comment by andyg0808** [comment]
@danielbeardsley, I didn't either. Thanks, @ianrohde, for pointing me at past-Andrew's explanation :laughing:.

File: bin/build.js (hunk 1 above)
Comment (andyg0808) [reply bin/build.js:10]: @danielbeardsley I think you're probably right, but that if we're assuming camel-case names, all-upper-case would be an acronym, and it's not completely unreasonable to turn that into a completely-hyphenated name for kebab-case.

@ianrohde Might be good to add a comment saying where that code came from.

Comment (ianrohde) [reply bin/build.js:11]: https://github.com/iFixit/core-primitives/pull/29/commits/1a423a041e4d560ab0f4e8f29f82b65916aaf5b0

**Review by andyg0808 — APPROVED** [review]

**Comment by andyg0808** [comment]
CR :goat: through 1a423a0

**Comment by danielbeardsley** [comment]
CR 😀

**Pull request merged**